checks.d/haproxy.py

```diff
@@ -287,9 +295,21 @@ def _tag_match_patterns(self, tag, filters):
                 return True
         return False
 
+    @staticmethod
+    def _normalize_status(status):
+        """
+        Try to match the HAProxy status with one of the statuses defined in `ALL_STATUSES`,
+        if it can't be matched return the status as-is in a tag-friendly format
+        """
+        formatted_status = status.lower().replace(" ", "_")
```

Comment:
Not sure if relevant, but I found this page because I'm failing with
```
  File "/opt/datadog-agent/agent/checks.d/haproxy.py", line 355, in _process_status_metric
    statuses_counter[tuple(tags)][counter_status] += count
KeyError: 'no check'
```

- so status can definitely be "no check".